src/block/block_map.c

```diff
@@ -58,12 +50,15 @@ __wt_block_map(
 
 	/*
 	 * Map the file into memory.
-	 * Ignore errors, we'll read the file through the cache if map fails.
+	 * Ignore not-supported errors, we'll read the file through the cache
+	 * if map fails.
 	 */
-	(void)__wt_mmap(session, block->fh, mapp, maplenp, mappingcookie);
+	ret = block->fh->fh_map(session, block->fh, mapp, maplenp);
+	if (ret == ENOTSUP)
```

Comment:
@agorrod 
> I don't believe checking ENOTSUP is safe across platforms. We don't translate error codes from OS calls to be POSIX compliant.

In these calls, I'm checking for specific returns from the WiredTiger Windows code, not OS calls, for example, the Windows map-preload code looks like this:
```
/*
 * __wt_win_map_preload --
 *      Cause a section of a memory map to be faulted in.
 */
int
__wt_win_map_preload(
    WT_SESSION_IMPL *session, WT_FH *fh, const void *p, size_t size)
{
        WT_UNUSED(session);
        WT_UNUSED(fh);
        WT_UNUSED(p);
        WT_UNUSED(size);

        return (ENOTSUP);
}
```
I think that's safe, unless the POSIX and Windows error namespace overlaps and a different error from a Windows call could look like `ENOTSUP`.